### PR TITLE
Test jackson support for scala enumerations

### PR DESCRIPTION
**Pull request by ignasi35**
Akka Jackson support may be used to read data produced by Play-JSON and other libraries.
Scala Enumerations use different default formats. Given this Enumeration and instance:

```scala
object InnerPlanet extends Enumeration {
  type InnerPlanet = Value
  val Mercury, Venus, Earth, Mars = Value
}
val galacticTrip = Seq(Earth, Mars, Venus)
```

Play JSON produces: `["Earth", "Mars", "Venus"]` (a `JsString` per value)

While Jackson still defaults to a legacy format: (formatting mine)

```
[
 {"enumClass":"example.InnerPlanet","value":"Earth"},
 {"enumClass":"example.InnerPlanet","value":"Mars"},
 {"enumClass":"example.InnerPlanet","value":"Venus"},
]
```

Using Jackson's `JsonScalaEnumeration` it is possible to produce `["Earth", "Mars", "Venus"]`.

This PR introduces tests for both formats.

### CODE REVIEW

**Comment by ignasi35** [comment]
This change provides not a huge value, and is more an example to exercise an edge case. Happy to discard this PR, TBH.

File: akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala [hunk 1]
```diff
@@ -96,6 +97,15 @@ object ScalaTestMessages {
...
   final case class OldCommandNotInBindings(name: String)
 
+  object Planet extends Enumeration {
+    type Planet = Value
+    val Mercury, Venus, Earth, Mars, Krypton = Value
+  }
+  final case class Alien(name:String, planet:Planet.Planet) extends TestMessage
+
+  class PlanetType extends TypeReference[Planet.type] {}
+  final case class Superhero(name:String, @JsonScalaEnumeration(classOf[PlanetType]) planet:Planet.Planet) extends TestMessage
```
Comment (patriknw) [inline akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala]: is it possible to enable the string value format by default, using class names in the format should be avoided

Comment (ignasi35) [reply akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala]: It's not possible in Jackson 2.2 but `It is expected this behavior will be deprecated in version 2.3, and removed entirely in 3.0.`( [ref](https://github.com/FasterXML/jackson-module-scala/wiki/Enumerations)). The annotation is used to provide the `class PlanetType extends TypeReference[Planet.type]` information that's missing when not adding the classname in the serialized form.

It'd be great if a setting enabled a compile or runtime check to ensure only the single-String format (based on the metadata provided by `@JsonScalaEnumeration`) is used.

I guess we could add a small mention in the docs?

Comment (patriknw) [reply akka-serialization-jackson/src/test/scala/akka/serialization/jackson/JacksonSerializerSpec.scala]: yes, we have a section Annotations where we could mention such gotcha
https://doc.akka.io/docs/akka/current/serialization-jackson.html#annotations

**Comment by ignasi35** [comment]
@patriknw I've added a section in the docs.

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]
LGTM
